src/main/java/mate/academy/spring/service/impl/CustomUserDetailService.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.spring.service.impl;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.findByEmail(email).orElseThrow(() ->
+                new UsernameNotFoundException("Can't find user with " + "email: " + email));
+        UserBuilder builder = withUsername(email);
+        builder.password(user.getPassword());
+        builder.authorities(user.getRoles().stream()
```

Comment:
```suggestion
        builder.roles(user.getRoles().stream()
```
or your can use authorities() but add "ROLE_" prefix for each role name